ui/app/components/app/transaction-time-remaining/transaction-time-remaining.component.js

```diff
@@ -0,0 +1,48 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import {
+  SUBMITTED_STATUS,
+} from '../../../helpers/constants/transactions'
+
+const statusToTextHash = {
+  [SUBMITTED_STATUS]: 'pending',
+}
+
+export default class TransactionTimeRemaining extends PureComponent {
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    statusKey: PropTypes.string,
+    className: PropTypes.string,
+    currentTimeEstimate: PropTypes.string,
+    fetchBasicGasAndTimeEstimates: PropTypes.func,
+    fetchGasEstimates: PropTypes.func,
+    blockTime: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+    ]),
+  }
+
+  render () {
+    const { className, statusKey } = this.props
+
+    if (statusToTextHash[statusKey] === 'pending') {
+      const promise = this.props.fetchBasicGasAndTimeEstimates()
```

Comment:
Is there a reason this was put in the `render` method as opposed to `componentDidMount`?

Reply:
I also wonder if this should be done here, or somewhere higher in the component hierarchy. If it is called in this component, it will be called for every pending transaction. But if it were called in `metamask-extension/ui/app/components/app/transaction-list/transaction-list.component.js` for instance, it would only be called once for the entire list.

Reply:
I could not get `statusKey` to resolve during `componentDidMount`  and thats why I wrote it on `render` method